src/bgw/scheduler.c

```diff
@@ -311,9 +307,12 @@ on_failure_to_start_job(ScheduledBgwJob *sjob)
 {
 	StartTransactionCommand();
 	if (!ts_bgw_job_get_share_lock(sjob->job.fd.id, CurrentMemoryContext))
+	{
 		elog(WARNING,
 			 "scheduler detected that job %d was deleted while failing to start",
 			 sjob->job.fd.id);
+		ts_bgw_job_cache_invalidate_callback();
```

Comment:
Nit: Still think it makes sense to move cache invalidation into `ts_bgw_job_get_share_lock()` to avoid this repetition (and risk forgetting to invalidate the cache in case of a missing job when this function is called in other places)